tensorpipe/test/channel/channel_test.h

```diff
@@ -203,39 +203,51 @@ class ChannelTestCase {
 
 template <typename TBuffer>
 class ClientServerChannelTestCase : public ChannelTestCase<TBuffer> {
-  std::vector<std::future<std::shared_ptr<tensorpipe::transport::Connection>>>
-  accept(
-      std::shared_ptr<tensorpipe::transport::Listener> listener,
-      size_t numConnections) {
-    auto promises = std::make_shared<std::vector<
-        std::promise<std::shared_ptr<tensorpipe::transport::Connection>>>>(
+  std::future<std::pair<
+      tensorpipe::Error,
+      std::vector<std::shared_ptr<tensorpipe::transport::Connection>>>>
+  accept(tensorpipe::transport::Listener& listener, size_t numConnections) {
+    auto promise = std::make_shared<std::promise<std::pair<
+        tensorpipe::Error,
+        std::vector<std::shared_ptr<tensorpipe::transport::Connection>>>>>();
+    auto connections = std::make_shared<
+        std::vector<std::shared_ptr<tensorpipe::transport::Connection>>>(
         numConnections);
+    auto connectionsMutex = std::make_shared<std::mutex>();
```

Comment:
Nit: A similar comment applies here, we're creating three separate shared_ptrs, which we then however always pass around together. It may make sense to "bundle" these values into a single struct and allocate a single shared_ptr.